src/Control/Bind.js

```diff
@@ -1,9 +1,21 @@
-export const arrayBind = function (arr) {
-  return function (f) {
-    var result = [];
-    for (var i = 0, l = arr.length; i < l; i++) {
-      Array.prototype.push.apply(result, f(arr[i]));
+export const arrayBind =
+  typeof Array.prototype.flatMap === "function"
+    ? function (arr) {
+      return function (f) {
+        return arr.flatMap(f);
+      };
     }
-    return result;
-  };
-};
+    : function (arr) {
+      return function (f) {
+        const result = [];
+        const l = arr.length;
+        for (var i = 0; i < l; i++) {
+          const xs = f(arr[i]);
+          const k = xs.length;
+          for (var j = 0; j < k; j++) {
+            result.push(xs[j]);
+          }
+        }
+        return result;
+      };
+    };
```

Comment:
Maybe worth pulling this out to a standalone function, to export and test

Reply:
I don't think a legacy FFI implementation should be a public Prelude export. If we want to test it we should just duplicate it, for now, and maybe add a comment.